collections_hierarchy/mixins.py

```diff
@@ -1,87 +1,103 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        for x, y in zip(self, other):
+            if not x == y:
+                pass
+            else:
+                return self == other
 
     def __ne__(self, other):
-        return not self == other
+        for x, y in zip(self, other):
+            if x == y:
+                pass
+            else:
+                return not self == other
 
     def __lt__(self, other):
-        pass
+        return self < other
 
     def __gt__(self, other):
-        pass
+        return self > other
 
     def __le__(self, other):
-        pass
+        return self <= other
 
     def __ge__(self, other):
-        pass
+        return self >= other
 
 
 class SequenceMixin(object):
     def __iter__(self):
-        pass
+        self.n = 0
+        self.elements = self.get_elements()
+        return(self)
 
     def __next__(self):
-        pass
+        while self.n < len(self.elements):
+            element = self.elements[self.n]
+            self.n += 1
+            return element
 
+        raise StopIteration
+            
     next = __next__
 
     def __len__(self):
-        pass
+        return len(self)
```

Comment:
The next three methods have the same issue.  The `return` statement is actually referencing the method itself.  Once you get down to `__contains__()`, though, you're good, because you're not operating on self directly, but on some other object.